workspace/TS100/Core/BSP/Pine64/I2C_Wrapper.cpp

```diff
@@ -10,523 +10,532 @@
-          timeout = 0;
-          state = I2C_START;
-          in_rx_cycle = 0;
-        }
-      } else {
-        /* one byte master reception procedure (polling) */
-        if (number_of_byte < 2) {
-          /* disable acknowledge */
-          i2c_ack_config(I2C0, I2C_ACK_DISABLE);
-          /* clear ADDSEND register by reading I2C_STAT0 then I2C_STAT1 register
-           * (I2C_STAT0 has already been read) */
-          i2c_flag_get(I2C0, I2C_FLAG_ADDSEND);
-          /* send a stop condition to I2C bus*/
-          i2c_stop_on_bus(I2C0);
-          /* wait for the byte to be received */
-          while (!i2c_flag_get(I2C0, I2C_FLAG_RBNE))
-            ;
-          /* read the byte received from the EEPROM */
-          *p_buffer = i2c_data_receive(I2C0);
-          /* decrement the read bytes counter */
-          number_of_byte--;
-          timeout = 0;
-        } else { /* more than one byte master reception procedure (DMA) */
-          dma_deinit(DMA0, DMA_CH6);
-          dma_init_struct.direction = DMA_PERIPHERAL_TO_MEMORY;
-          dma_init_struct.memory_addr = (uint32_t)p_buffer;
-          dma_init_struct.memory_inc = DMA_MEMORY_INCREASE_ENABLE;
-          dma_init_struct.memory_width = DMA_MEMORY_WIDTH_8BIT;
-          dma_init_struct.number = number_of_byte;
-          dma_init_struct.periph_addr = (uint32_t)&I2C_DATA(I2C0);
-          dma_init_struct.periph_inc = DMA_PERIPH_INCREASE_DISABLE;
-          dma_init_struct.periph_width = DMA_PERIPHERAL_WIDTH_8BIT;
-          dma_init_struct.priority = DMA_PRIORITY_ULTRA_HIGH;
-          dma_init(DMA0, DMA_CH6, &dma_init_struct);
+				/* enable acknowledge */
+				i2c_ack_config(I2C0, I2C_ACK_ENABLE);
+				/* i2c master sends start signal only when the bus is idle */
+				while (i2c_flag_get(I2C0, I2C_FLAG_I2CBSY)
+						&& (timeout < I2C_TIME_OUT )) {
+					timeout++;
+				}
+				if (timeout < I2C_TIME_OUT) {
+					/* send the start signal */
+					i2c_start_on_bus(I2C0);
+					timeout = 0;
+					state = I2C_SEND_ADDRESS;
+				} else {
+					I2C_Unstick();
+					timeout = 0;
+					state = I2C_START;
+				}
+			} else {
+				i2c_start_on_bus(I2C0);
+				timeout = 0;
+				state = I2C_SEND_ADDRESS;
+			}
+			break;
+		case I2C_SEND_ADDRESS:
+			/* i2c master sends START signal successfully */
+			while ((!i2c_flag_get(I2C0, I2C_FLAG_SBSEND))
+					&& (timeout < I2C_TIME_OUT )) {
+				timeout++;
+			}
+			if (timeout < I2C_TIME_OUT) {
+				if (RESET == in_rx_cycle) {
+					i2c_master_addressing(I2C0, DevAddress, I2C_TRANSMITTER);
+					state = I2C_CLEAR_ADDRESS_FLAG;
+				} else {
+					i2c_master_addressing(I2C0, DevAddress, I2C_RECEIVER);
+					state = I2C_CLEAR_ADDRESS_FLAG;
+				}
+				timeout = 0;
+			} else {
+				timeout = 0;
+				state = I2C_START;
+				in_rx_cycle = 0;
+			}
+			break;
+		case I2C_CLEAR_ADDRESS_FLAG:
+			/* address flag set means i2c slave sends ACK */
+			while ((!i2c_flag_get(I2C0, I2C_FLAG_ADDSEND))
+					&& (timeout < I2C_TIME_OUT )) {

```

Comment:
Trailing whitespace after `I2C_TIME_OUT` before `))`? (Actually happens later as well, in same context)

Reply:
To be honest I'm just using the default formatter in the Nuclei IDE (eclipse) so unsure why the extra space is there.. hmm.